hypershift-operator/metrics.go

```diff
@@ -156,52 +210,68 @@ func (m *hypershiftMetrics) collect(ctx context.Context) error {
 	if err := m.observeNodePools(ctx, &nodePools); err != nil {
 		return err
 	}
+
 	return nil
 }
 
 func setupMetrics(mgr manager.Manager) error {
-	metrics := newMetrics(mgr.GetClient(), mgr.GetLogger().WithName("metrics"))
+	var hypershiftImage ImageInfo
+
+	// We need to create a new client because the manager one still does not have the cache started
+	tmpClient, err := crclient.New(mgr.GetConfig(), crclient.Options{Scheme: hyperapi.Scheme})
+	if err != nil {
+		return fmt.Errorf("error creating a temporary client: %w", err)
+	}
+	// Grabbing the Image and ImageID from Operator
+	if hypershiftImage.image, hypershiftImage.imageId, err = getOperatorImage(tmpClient); err != nil {
+		fmt.Printf("pod not found, reporting empty image")
```

Comment:
you still need to check that the error is notFound here.
If !notFound then return error
If notFound then please use the logger to print this message.